### ISSUE TITLE
Periodic backup failure deletes existing backup file

### ISSUE THREAD

**Issue by Terrance** [KieronQuinn/AmbientMusicMod#181]
If the periodic backup fails (e.g. because of "Failed to connect to Now Playing to backup settings"), the previous day's backup is lost -- `amm_backup_periodic.ammbkp` is a zero-sized file.

It would be safer to create a temporary file for the new backup, and replace the old one on success.

**Comment by KieronQuinn** [comment]
2.3.2 now handles this correctly and will revert back if writing fails

**Comment by Terrance** [comment]
Running 2.3.2, I'm not sure this is working as expected:

```
$ ls -l Backups/Now\ Playing/ | grep periodic
-rw-rw---- 1 root everybody       0 Feb 10 03:00 amm_backup_periodic.ammbkp
-rw-rw---- 1 root everybody  832020 Feb  5 03:00 amm_backup_periodic.ammbkp (1).tmp
-rw-rw---- 1 root everybody       0 Feb  9 03:00 amm_backup_periodic.ammbkp (10).tmp
-rw-rw---- 1 root everybody       0 Feb 10 03:00 amm_backup_periodic.ammbkp (11).tmp
-rw-rw---- 1 root everybody       0 Feb  5 03:00 amm_backup_periodic.ammbkp (2).tmp
-rw-rw---- 1 root everybody  835414 Feb  6 03:00 amm_backup_periodic.ammbkp (3).tmp
-rw-rw---- 1 root everybody       0 Feb  6 03:00 amm_backup_periodic.ammbkp (4).tmp
-rw-rw---- 1 root everybody  838424 Feb  7 03:00 amm_backup_periodic.ammbkp (5).tmp
-rw-rw---- 1 root everybody       0 Feb  7 03:00 amm_backup_periodic.ammbkp (6).tmp
-rw-rw---- 1 root everybody  843741 Feb  8 03:00 amm_backup_periodic.ammbkp (7).tmp
-rw-rw---- 1 root everybody       0 Feb  8 03:00 amm_backup_periodic.ammbkp (8).tmp
-rw-rw---- 1 root everybody  848718 Feb  9 03:00 amm_backup_periodic.ammbkp (9).tmp
-rw-rw---- 1 root everybody  830614 Feb  4 03:00 amm_backup_periodic.ammbkp.tmp
```

**Comment by KieronQuinn** [comment]
This was fixed in 2.3.3, delete the duplicates.
